zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/DataSetInfo.scala

```diff
@@ -45,14 +46,18 @@ object DataSetInfo {
 
         val schema = dataSetInfo.schema
         val primaryKey = schema.primaryKey.map(schema.getField(_).get)
-        val timeField = schema.getField(schema.timeField) match {
-          case Some(f) if f.isInstanceOf[TimeField] => f.asInstanceOf[TimeField]
-          case None if schema.timeField.isEmpty => throw new QueryParsingException(s"Time field is not specified for ${schema.typeName}.")
-          case _ => throw new QueryParsingException(s"${schema.timeField} is not a valid time field.")
+        val resolvedSchema = schema.timeField match {
+          case Some(field) =>
```

Comment:
I think I saw the exactly same code snippet before. Could we unify them as a single method?